### PR TITLE
feat(proxy): Add simple proxy configuration (proxyHost/proxyPort)

### PR DESCRIPTION
**Pull request by patrice-conil**
This PR add 2 new configuration fields to be able to set proxy-host and proxy-port in application.properties.

### CODE REVIEW

File: integration-tests/vault/src/test/resources/application-proxy.properties [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+quarkus.vault.log-confidentiality-level=low
+quarkus.vault.renew-grace-period=10
+
+quarkus.log.category."io.quarkus.vault".level=DEBUG
+
+#quarkus.log.level=DEBUG
+#quarkus.log.console.level=DEBUG
+
+quarkus.vault.proxyHost=localhost
```
Comment (vsevel) [inline integration-tests/vault/src/test/resources/application-proxy.properties:24]: could be `proxy-host` for consistency

File: runtime/src/main/java/io/quarkus/vault/runtime/config/VaultBootstrapConfig.java [hunk 2]
```diff
@@ -228,6 +228,18 @@ public class VaultBootstrapConfig {
...
+     * The proxy host. If set the client is configured to use a proxy.
+     */
+    @ConfigItem
+    public Optional<String> proxyHost;
+
+    /**
+     * The port the proxy is listening on, 3128 by default.
+     */
+    @ConfigItem(defaultValue = "3128")
+    public Optional<Integer> proxyPort;
```
Comment (vsevel) [inline runtime/src/main/java/io/quarkus/vault/runtime/config/VaultBootstrapConfig.java:241]: if there is a default value, why would we make it optional?

**Review by vsevel — CHANGES_REQUESTED** [review]

File: integration-tests/vault/src/test/resources/application-proxy.properties (hunk 1 above)
Comment (patrice-conil) [reply integration-tests/vault/src/test/resources/application-proxy.properties]: Done in new commit.

File: runtime/src/main/java/io/quarkus/vault/runtime/config/VaultBootstrapConfig.java (hunk 2 above)
Comment (patrice-conil) [reply runtime/src/main/java/io/quarkus/vault/runtime/config/VaultBootstrapConfig.java]: Good remark ... also integrated in new commit

**Comment by vsevel** [comment]
can you squash your commits?

**Comment by patrice-conil** [comment]
Squash done ... I hope

**Comment by patrice-conil** [comment]
I'm trying to add a wiremock instance to really validate proxying in integration test.

**Comment by patrice-conil** [comment]
Hi @vsevel, job done ... ready for review

**Review by vsevel — APPROVED** [review]
looks good to me 
need to have a look @kdubb ?

**Comment by vsevel** [comment]
thanks @patrice-conil

**Pull request merged**